docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -806,6 +806,183 @@ One challenge when developing locally is that a developer might make modificatio
+
+Depending on the CMS you will also need to store the preview webhook URL. This might also be stored in the plugin extension configuration, but often is stored in a separate webhooks configuration. [Find out how to get that webhook url here](https://support.gatsbyjs.com/hc/en-us/articles/360052324394-Build-and-Preview-Webhooks)
+
+NOTE: The Content Sync URL can be found in the same place as the webhook url.
+
+##### Building up the manifest id
+
+Recall that we need to create a matching manifest id in the CMS AND the Gatsby plugin. Whenever content is saved we can build up a new manifest id that will look the same as the manifest id we created in the source plugin
+
+```javascript
+const manifestId = `${project.id}-${entryItem.id}-${updatedAt}`
+```
+
+In the CMS extension, we should have access to
+
+- the project id (if the CMS uses one)
+- the content id
+- the timestamp that the content was updated at (or some other piece of data that is tied to a very specific state of saved content)
+
+##### Starting the preview build (optional)
+
+If the CMS does not handle this part automatically we will need to tell Gatsby cloud to build a preview by `POST`ing to the Gatsby Cloud preview build webhook url.
+
+##### Opening the Content Sync waiting room
+
+Once we've built a `manifestId` and `POST`ed to the preview build webhook url, we need to open a new tab/window with a modified version of the Content Sync URL. You get that by grabbing the Content Sync URL you stored in the CMS extension earlier and appending the Gatsby source plugin name and the content's `manifestId` that you just created, `https://gatsbyjs.com/content-sync/<siteId>/<sourcePluginName>/<manifestId>`.
+
+#### Useful Content Sync development tips
+
+Here are some things to keep in mind and some "gotchas" depending on how the CMS acts.
+
+- Sometimes you will need to wait to make sure you have the correct `updatedAt` timestamp as some CMS may take a second to update their backend and then wait for the change to propagate to the frontend. While others will immediately update the frontend and then propagate that to the backend.
+- Make sure that a preview webhook is being sent to Gatsby Cloud after the content is edited, whether it's before you press the "Open Preview" button or the "Open Preview" is the trigger that sends the webhook.
+- While developing, you can set the Gatsby `VERBOSE` env variable to `"true"` to see additional logs that will help you debug what's happening in the source plugin.
+- When you click the "Open Preview" button in the CMS the `manifestId` in the URL should match the `manifestId` that the source plugin creates from that revision.
+- The node manifests get written out in the `public` dir of your gatsby site, so you can check to manifests on your local disk `/public/<sourcePluginName>/<manifestId>` or you can navigate directly to that piece of content `https://<your-domain>/<sourcePluginName>/<manifestId>`
```

Comment:
It should be `/public/__node-manifests/<sourcePluginName>/<manifestId>.json` and `https://<your-domain>/__node-manifests/<sourcePluginName>/<manifestId>`